src/cmd/generate/generate_e2e_test.go

```diff
@@ -0,0 +1,194 @@
+		}
+
+		dir, err := os.ReadDir(rev4YamlPath)
+		if err != nil {
+			t.Fatal(err)
+		}
+		for _, file := range dir {
+
+			bytes, err := os.ReadFile(rev4YamlPath + file.Name())
+			if err != nil {
+				t.Fatal(err)
+			}
+			oscalDoc := map[string]interface{}{}
+			err = yaml.Unmarshal(bytes, &oscalDoc)
+			if err != nil {
+				t.Fatal(err)
+			}
+			ValidateKeys(oscalDoc, string(typeBytes), t)
+		}
+	})
+
+	t.Run("Rev5", func(t *testing.T) {
+		t.Parallel()
+		typeBytes, err := os.ReadFile(oscal111Types)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		dir, err := os.ReadDir(rev5YamlPath)
+		if err != nil {
+			t.Fatal(err)
+		}
+		for _, file := range dir {
+
+			bytes, err := os.ReadFile(rev5YamlPath + file.Name())
+			if err != nil {
+				t.Fatal(err)
+			}
+			oscalDoc := map[string]interface{}{}
+			err = yaml.Unmarshal(bytes, &oscalDoc)
+			if err != nil {
+				t.Fatal(err)
+			}
+			ValidateKeys(oscalDoc, string(typeBytes), t)
+		}
+
+	})
+}
+
+func TestFieldStability(t *testing.T) {
+	t.Parallel()
+
+	t.Run("Rev4", func(t *testing.T) {
+		t.Parallel()
+		dir, err := os.ReadDir(rev4YamlPath)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		for _, file := range dir {
+
+			bytes, err := os.ReadFile(rev4YamlPath + file.Name())
+			if err != nil {
+				t.Fatal(err)
+			}
+			oscalDoc := oscalTypes_1_0_4.OscalCompleteSchema{}
+			err = yaml.Unmarshal(bytes, &oscalDoc)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// Marshal the document back to yaml
+			marshaled, err := yaml.Marshal(oscalDoc)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			actual := map[string]interface{}{}
+			expected := map[string]interface{}{}
+			err = yaml.Unmarshal(marshaled, &actual)
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = yaml.Unmarshal(bytes, &expected)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if !reflect.DeepEqual(actual, expected) {
+				t.Error("expected marshaled yaml to be equal to the original yaml")
+			}
+
+		}
+	})
+
+	t.Run("Rev5", func(t *testing.T) {
+		t.Parallel()
+		dir, err := os.ReadDir(rev5YamlPath)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		for _, file := range dir {
+
+			bytes, err := os.ReadFile(rev5YamlPath + file.Name())
+			if err != nil {
+				t.Fatal(err)
+			}
+			oscalDoc := oscalTypes_1_1_1.OscalCompleteSchema{}
+			err = yaml.Unmarshal(bytes, &oscalDoc)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// Marshal the document back to yaml
+			marshaled, err := yaml.Marshal(oscalDoc)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			actual := map[string]interface{}{}
+			expected := map[string]interface{}{}
+			err = yaml.Unmarshal(marshaled, &actual)
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = yaml.Unmarshal(bytes, &expected)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			if !reflect.DeepEqual(actual, expected) {
+				t.Error("expected marshaled yaml to be equal to the original yaml")
+			}
+
+		}
+	})
+}
+
+func ValidateKeys(model map[string]interface{}, typeString string, t *testing.T) {
```

Comment:
How does this function account for testing of a specific model and the structs that were separated due to being unique?

IE from the `CHANGELOG.md`:
```
update `PlanOfActionAndMilestones.LocalDefinitions` is now of type `PlanOfActionAndMilestonesLocalDefinitions`
```

Does it account for these? does that even matter? My understanding is that it's looking for `local-definitions` in this example - which it will find multiple instances of - but will it find the right match?

Reply:
I don't really think it matters, those tests are at the unit level that tests the logic for handling duplicates and creating the type. Honestly I am not really sure how that would be testable because at this stage the logic for naming and generating has already been tested. I am seeing this more as a way to catch a field that may be missed in future updates if the way the schema is built changes.

Reply:
Definitely agree with the premise of catching something here that wasn't caught/tested elsewhere.

My comment is more about the accuracy of the check - take the `local-definitions` object for example - Line 175 is simply performing a `string.Contains()` check. We know that this tag will exist in multiple locations for unique objects. Plus we know with context that any match will return true. Do we care? worth annotating in comments to what level of granularity to expect in the function.

All in all, I'm fine with this being a piece of evidence that works to augment other testing that we perform, but i'd like to capture intent/scope in comments in the code.